src/platform/packages/private/kbn-apm-config-loader/src/filters/pii_filter.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { FilterFn } from 'elastic-apm-node';
+import { isPlainObject, has } from 'lodash';
+
+export const piiFilter: FilterFn = (payload) => {
+  // if user context is defined, apply the pii filter
+  if (has(payload.context, 'user')) {
+    // if the user context has loopable properties, redact the values
+    if (payload.context.user != null && isPlainObject(payload.context.user)) {
+      const userContextKeys = Object.keys(payload.context.user);
+      for (const key of userContextKeys) {
+        payload.context.user[key] = '[REDACTED]';
+      }
```

Comment:
nit: I wonder if, for the sake of safety, it makes sense to try...catch this block when the catch overrides the entire user object like in the `else`.

Although I can't think of any case where this would break... (maybe if the user object is `Object.freeze`?)